core/trino-main/src/main/java/io/trino/connector/ConnectorAwareNodeManager.java

```diff
@@ -28,27 +29,28 @@
     private final InternalNodeManager nodeManager;
     private final String environment;
     private final CatalogName catalogName;
+    private final boolean schedulerIncludeCoordinator;
 
-    public ConnectorAwareNodeManager(InternalNodeManager nodeManager, String environment, CatalogName catalogName)
+    public ConnectorAwareNodeManager(InternalNodeManager nodeManager, String environment, CatalogName catalogName, boolean schedulerIncludeCoordinator)
     {
         this.nodeManager = requireNonNull(nodeManager, "nodeManager is null");
         this.environment = requireNonNull(environment, "environment is null");
         this.catalogName = requireNonNull(catalogName, "catalogName is null");
+        this.schedulerIncludeCoordinator = schedulerIncludeCoordinator;
     }
 
     @Override
     public Set<Node> getAllNodes()
     {
-        return ImmutableSet.<Node>builder()
-                .addAll(getWorkerNodes())
-                .addAll(nodeManager.getCoordinators())
-                .build();
+        return ImmutableSet.copyOf(nodeManager.getActiveConnectorNodes(catalogName));
```

Comment:
`ImmutableSet.copyOf` is redundant, it's NodeManager's responsibility to provide immutability

```suggestion
        return nodeManager.getActiveConnectorNodes(catalogName);
```

(being fixed in https://github.com/trinodb/trino/pull/7021)

Reply:
This is actually required because `getActiveConnectorNodes` returns `Set<InternalNode>` and we need `Set<Node>` here